pkg/grafana/dashboard_test.go

```diff
@@ -81,4 +81,59 @@ func TestDashboard(t *testing.T) {
 	})
 
 	_ = os.Unsetenv("GRAFANA_URL")
+
+	t.Run("Validate/Prepare", func(t *testing.T) {
+		handler := DashboardHandler{}
+		tests := []struct {
+			Name                string
+			Resource            grizzly.Resource
+			ValidateErrorString string
+			ExpectedUID         string
+		}{
+			{
+				Name:                "name and UID match",
+				Resource:            grizzly.NewResource("apiVersion", "Dashboard", "name1", map[string]any{"uid": "name1"}),
+				ValidateErrorString: "",
+				ExpectedUID:         "name1",
+			},
+			{
+				Name:                "no UID provided",
+				Resource:            grizzly.NewResource("apiVersion", "Dashboard", "name1", map[string]any{}),
+				ValidateErrorString: "",
+				ExpectedUID:         "name1",
+			},
+			{
+				Name:                "name and UID differ",
+				Resource:            grizzly.NewResource("apiVersion", "Dashboard", "name1", map[string]any{"uid": "name2"}),
+				ValidateErrorString: "uid 'name2' and name 'name1', don't match",
+			},
+			{
+				Name:                "no name provided",
+				Resource:            grizzly.NewResource("apiVersion", "Dashboard", "", map[string]any{"uid": "name1"}),
+				ValidateErrorString: "name cannot be blank",
+			},
+			{
+				Name:                "neither name nor UID provided",
+				Resource:            grizzly.NewResource("apiVersion", "Dashboard", "", map[string]any{}),
+				ValidateErrorString: "name cannot be blank",
+			},
+		}
+		for _, test := range tests {
+			t.Run(test.Name, func(t *testing.T) {
+				err := handler.Validate(test.Resource)
+				if err != nil {
+					if test.ValidateErrorString == "" {
+						require.NoError(t, err)
+					} else {
+						require.Contains(t, err.Error(), test.ValidateErrorString)
+					}
+				} else {
+					require.NoError(t, err)
+					newResource := handler.Prepare(nil, test.Resource)
+					uid, _ := handler.GetUID(*newResource)
+					require.Equal(t, uid, test.ExpectedUID)
+				}
```

Comment:
```suggestion
				err := handler.Validate(test.Resource)
if test.ValidateErrorString != "" {
	require.Contains(t, err.Error(), test.ValidateErrorString)
	return
}

require.NoError(t, err)
newResource := handler.Prepare(nil, test.Resource)
uid, _ := handler.GetUID(*newResource)
require.Equal(t, uid, test.ExpectedUID)
```

I think you can remove a level of `if` here

Reply:
thx. Makes sense.